cli/src/cli.c

```diff
@@ -843,9 +843,18 @@ main(int argc, char *argv[])
     if (!global_rpc)
         goto out;
 
-    global_quotad_rpc = cli_quotad_clnt_rpc_init();
-    if (!global_quotad_rpc)
-        goto out;
+    /*
+     * Now, one doesn't need to initialize global rpc
+     * for quota unless and until quota is enabled.
+     * So why not put a check to save all the rpc related
+     * ops here.
+     */
+    ret = sys_access("/var/run/gluster/quotad/quotad.pid", F_OK);
```

Comment:
I think instead of checking pid you should check existence of socket file(/var/run/gluster/quotad.socket)

Reply:
I checked the scenario, it seems like the quotad.socket file is not deleted after quota is disabled and it is still present in the /var/lib/gluster directory. Hence even if quota is disabled after use, we'd continue to use the global rpc for quota in cli.

While the pid file is present only when quota is enabled and is then removed when quota is disabled.

Reply:
GLUSTERD_VAR_RUN_DIR should be used instead of "/var/run"

Reply:
I saw that the quotad.socket was referenced in the same manner in cli code and hence went along the convention. Will change the other references too if required to be made standard.

Reply:
@srijan-sivakumar , I don't think the above comment has been addressed in the latest commit. Can you please confirm once?

Reply:
@nik-redhat  So, the reason I added a new macro was that to use the GLUSTERD_VAR_RUN_DIR I saw we needed to have a call to snprintf and that required a couple of local variables being declared to arrive at the desired string ( i.e. /var/run/gluster/quotad/quotad.pid". 

Now we do have two macros GLUSTERD_VAR_RUN_DIR (/var/run/gluster ) and GLUSTER_QUOTAD_RUN_DIR (/quotad) but it seemed to add more operations being performed to reach the desired result and a macro seemed the simpler option. Though I agree that it is similar to what I had sent out in patch 1. :p 

It's just I felt creating a variable for storing the path and then appending operations will be a small overhead as we are using it only once. Anyway..let me know your thoughts..

Reply:
Yeah, makes sense.